modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/item_actions/ActionsDropdown.js

```diff
@@ -18,168 +18,39 @@ import ClayIcon from '@clayui/icon';
+import {isLink} from './ItemActions';
+import {itemActionsBasePropTypes} from './types/index';
 
-const {MODAL_PERMISSIONS} = ACTION_ITEM_TARGETS;
-
-export function isLink(target, onClick) {
-	return !(target && target !== 'link') && !onClick;
-}
-
-export function handleAction(
-	{
-		confirmationMessage,
-		event,
-		itemId,
-		method,
-		onClick,
-		setLoading,
-		size,
-		successMessage,
-		target,
-		title,
-		url,
-	},
-	{
-		executeAsyncItemAction,
-		highlightItems,
-		openModal,
-		openSidePanel,
-		toggleItemInlineEdit,
-	}
-) {
-	if (confirmationMessage && !confirm(confirmationMessage)) {
-		return;
-	}
-
-	if (target?.includes('modal')) {
-		event.preventDefault();
-
-		if (target === MODAL_PERMISSIONS) {
-			openPermissionsModal(url);
-		}
-		else {
-			openModal({
-				size: resolveModalSize(target),
-				title,
-				url,
-			});
-		}
-	}
-	else if (target === 'sidePanel') {
-		event.preventDefault();
-
-		highlightItems([itemId]);
-		openSidePanel({
-			size: size || 'lg',
-			title,
-			url,
-		});
-	}
-	else if (target === 'async' || target === 'headless') {
-		event.preventDefault();
-
-		setLoading(true);
-		executeAsyncItemAction(url, method)
-			.then(() => {
-				openToast({
-					message:
-						successMessage ||
-						Liferay.Language.get('action-completed'),
-					type: 'success',
-				});
-				setLoading(false);
-			})
-			.catch((_) => {
-				setLoading(false);
-			});
-	}
-	else if (target === 'inlineEdit') {
-		event.preventDefault();
-
-		toggleItemInlineEdit(itemId);
-	}
-	else if (target === 'blank') {
-		event.preventDefault();
-
-		window.open(url);
-	}
-	else if (onClick) {
-		event.preventDefault();
-
-		event.target.setAttribute('onClick', onClick);
-		event.target.onclick();
-		event.target.removeAttribute('onClick');
-	}
-}
-
-function ActionItem({
+function DropdownItem({
 	action,
 	closeMenu,
-	handleAction,
 	itemData,
 	itemId,
-	method,
+	onActionClick,
 	onClick,
-	setLoading,
 	size,
-	title,
 	url,
 }) {
-	const context = useContext(DataSetContext);
-
-	const {data, icon, label, target} = action;
-
-	function handleClickOnLink(event) {
-		event.preventDefault();
-
-		handleAction(
-			{
-				confirmationMessage: data?.confirmationMessage,
-				event,
-				itemId,
-				method,
-				onClick,
-				setLoading,
-				size: size || 'lg',
-				successMessage: data?.successMessage,
-				target,
-				title,
-				url,
-			},
-			context
-		);
-
-		closeMenu();
-	}
-
-	const link = isLink(target, onClick);
-
-	const onActionDropdownItemClick = context.onActionDropdownItemClick;
+	const {icon, label, target} = action;
 
 	return (
 		<ClayDropDown.Item
-			href={link ? url : null}
-			onClick={(event) => {
-				if (onActionDropdownItemClick) {
-					onActionDropdownItemClick({
-						action,
-						event,
-						itemData,
-					});
-				}
-
-				if (!link) {
-					handleClickOnLink(event);
-				}
-			}}
+			href={isLink(target, onClick) ? url : null}
```

Comment:
`onClick` will never be set, we are only using `DropdownItem` in this file. We can definitely simplify this, not sure to what extent. Would simply `href={url}` work? :thinking: Or is something like `href={target === 'link' ? url : null}` necessary?

With this in mind, we can probably drop `onActionClick` and use `onClick` for its purpose.

Reply:
glad to remove some of the zombi code 👍